plugins/module_utils/network/sonic/utils/formatted_diff_utils.py

```diff
@@ -283,12 +334,12 @@ def derive_config_from_merged_cmd_dict(command, exist_conf, test_keys=None, key_
 
     key_matched = key_match_op(key_set, command, new_conf)
     if key_matched:
-        for key in trival_cmd_key_set:
-            new_conf[key] = command[key]
-
-        only_cmd_dict_list_key_set = dict_list_cmd_key_set.difference(common_dict_list_key_set)
-        for key in only_cmd_dict_list_key_set:
-            new_conf[key] = command[key]
+        done, new_conf = merge_op(key_set, command, new_conf)
+        if done:
+            return key_matched, new_conf
+        else:
+            nu, dict_list_exist_key_set = get_key_sets(new_conf)
+            common_dict_list_key_set = dict_list_cmd_key_set.intersection(dict_list_exist_key_set)
```

Comment:
`common_dict_list_key_set` is obtained earlier in line no: 333 and those keys are required to be processed individually.

Since in `merge_op`, keys (list and dict) that are not present in `exist_conf` are updated in `new_conf` the `common_dict_list_key_set` obtained in line no: 342 will also contain the new keys that were added in 'merge_op` which are not required to be processed again.
```suggestion
        done, new_conf = merge_op(key_set, command, new_conf)
        if done:
            return key_matched, new_conf
```

Reply:
In case of merge_op may modify new_conf, so re-calculate common_dict_list_key_set may needed.